### PR TITLE
[vNext][UX] Feature/mrtk interactable layout3D and InspectorFields

### PR DESCRIPTION
**Pull request by killerantz**
Overview
Moved  layout code out of Interactables and created a demo scene
Moved InspectorFields out of Interactables and created a demo scene
Moved some inspector helper code to InspectorUIUntility
Created receiver components to allow handling interactable state changes without Unity Events

### CODE REVIEW

File: Assets/MixedRealityToolkit-Examples/Demos/UX/Layout3D/Scripts/AutoCycletLayout3DSize.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Layout3D
+{
+    public class AutoCycletLayout3DSize : MonoBehaviour
```
Comment (cre8ivepark) [inline Assets/MixedRealityToolkit-Examples/Demos/UX/Layout3D/Scripts/AutoCycletLayout3DSize.cs]: Cyclet?

**Comment by radicalad** [comment]
@killerantz  this is super cool. Can we groupt these layout tools with ObjectCollection?

File: Assets/MixedRealityToolkit-Examples/Demos/UX/Layout3D/Scripts/AutoCycletLayout3DSize.cs (hunk 1 above)
Comment (killerantz) [reply Assets/MixedRealityToolkit-Examples/Demos/UX/Layout3D/Scripts/AutoCycletLayout3DSize.cs]: Thanks for catching that, I'll update shortly.

**Comment by killerantz** [comment]
@radicalad The grid would make sense in ObjectCollection for sure, the other items are more about aligning elements to build an object. I'm not sure most of them fit as a collection.

**Comment by cre8ivepark** [comment]
Release page shows 'Requires Unity 2018.1 or later' but Recommended version: 2018.2.14f1
Considering the fact that prefabs are broken when saved with 2018.2.x and opened with 2018.1.x, it would be great if we prepare these with 2018.1.x.

**Comment by StephenHodgson** [comment]
Most of the 2018.2.x builds don't work with the MRTK. I think recommending the latest for now is a good idea.

**Comment by SimonDarksideJ** [comment]
I've stated 2018.2.13f1 as a minimum in all my content so far. (which was the first 2018.2 build it actually worked.)
2018.2.1 - 2018.2.12 DO NOT WORK

**Comment by killerantz** [comment]
I fixed the prefabs, there's some property names that get changed in the prefab and scene files from 2018.1 to 2018.2, so if we update the prefabs, we should only do it in the earlier versions unless we are standardizing on the latest. Confirmed they work in 2018.1.9f2 and 2018.2.14f1

**Pull request merged**